### PR TITLE
DOC: optimize: change term zero to root

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

#17335
#### What does this implement/fix?

Changes documentation to refer to finding root rather than zeros of a function as discussed.

### CODE REVIEW

File: scipy/optimize/_zeros_py.py [hunk 1]
```diff
@@ -677,7 +677,7 @@ def brentq(f, a, b, args=(),
     """
     Find a root of a function in a bracketing interval using Brent's method.
 
-    Uses the classic Brent's method to find a zero of the function `f` on
+    Uses the classic Brent's method to find a roo of the function `f` on
```
Comment (mdhaber) [inline scipy/optimize/_zeros_py.py:680]:
```suggestion
    Uses the classic Brent's method to find a root of the function `f` on
```

**Review by mdhaber — APPROVED** [review]
Aside from the typo, everything looks good. It seems that all of these uses of zero did mean to refer to the value of the argument (the root) rather than the value of the function at the root.

**Comment by j-bowhay** [comment]
> Aside from the typo, everything looks good. It seems that all of these uses of zero did mean to refer to the value of the argument (the root) rather than the value of the function at the root.

Yes, the only occurrence I did not change was
https://github.com/scipy/scipy/blob/cdf1330c15ea1712580bfbd243de765c56bfc8a3/scipy/optimize/_zeros_py.py#L180-L181

File: scipy/optimize/__init__.py [hunk 2]
```diff
@@ -200,7 +200,8 @@
 
 .. seealso::
 
-   `scipy.optimize.cython_optimize` -- Typed Cython versions of zeros functions
+   `scipy.optimize.cython_optimize` -- Typed Cython versions of root finding
+   functions
```
Comment (mdhaber) [inline scipy/optimize/__init__.py:204]: Looks like this caused the refguide failure.

Comment (j-bowhay) [reply scipy/optimize/__init__.py]: Indeed, couldn't remember if you could split over multiple lines with this directive. Have fixed.

**Review by rgommers — APPROVED** [review]
LGTM too, thanks @j-bowhay and @mdhaber

**Pull request merged**